### ISSUE TITLE
Waiting for connection: no detailed status

### ISSUE THREAD

**Issue by MakisH** [precice/precice#444]
When establishing communication, the user does not get any details on how many partners (and which) it is waiting for:

```
(0) 12:59:10 [impl::SolverInterfaceImpl]:214 in initialize: Setting up master communication to coupling partner/s
(0) 12:59:22 [impl::SolverInterfaceImpl]:219 in initialize: Masters are connected
```

The last message is shown only after every participant is connected (even with logger set with `filter="%Severity% >= trace"`).

In a muli-participant scenario, it would be very helpful to get a message such as `Found participant Inner-Fluid` and then `Found participant Outer-Fluid`. This way, it is easier to debug a case that cannot start.

Even further, it would be nice to have a status of how many participants have been connected (e.g. `Found participant Inner-Fluid [1/2]` and then `Found participant Outer-Fluid [2/2]`). Or, instead of `[1/2]`, write `[1 participant pending]` or `[waiting for participant/s: Outer-Fluid]`.

Printing the address and protocol used to find the masters (together with own address) would also be helpful (at least in the debug/trace mode), as it would give hints for wrong configuration or old address files. 

Example:

```
Setting up master communication to coupling partner/s: Inner-Fluid, Outer-Fluid
My IP address is 1.2.3.5
Looking for participant Inner-Fluid using TCP Sockets at IP address 1.2.3.4
Looking for participant Outer-Fluid using TCP Sockets at IP address 1.2.3.6
Found participant Inner-Fluid
Found participant Outer-Fluid
Masters are connected
```
